File: c++ applications/JNALink/JNALink.cpp
#include "JNALink.h"

double getTime(){
    return simulator->getTime();
}
void performOneStepOfSimulation(){
    simulator->performOneStepOfSimulation();
}
void performWholeSimulation(){
    simulator->performWholeSimulation();
}
void setSimulatorForNewSimulation(int seed, char* settingsPath){
    simulator->setSimulatorForNewSimulation(seed,string(settingsPath));
}
double rval(int obs){
    return simulator->rval(obs);
}
double rvals(char* obs){
    return simulator->rval(string(obs));
}

void registerSimulator(SimulatorLink* sim){
    delete simulator;
    simulator = sim;
}

File: Multivesta meets c++/src/inventoryManagementTesting/JNALink.java
package inventoryManagementTesting;

import com.sun.jna.Library;
import com.sun.jna.Native;

public interface JNALink extends Library {
    JNALink INSTANCE = (JNALink) Native.load("InventoryManagementSimulator", JNALink.class);

    double getTime();
    void performOneStepOfSimulation();
    void performWholeSimulation();
    void setSimulatorForNewSimulation(int seed, String settingsPath);
    double rval(int obs);
    double rvals(String obs);
}
